### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by bctsui**
* Ensure proper exit of goroutines when job queue is closed.
* Removed chan of chan and consume directly from jobQueue.

### CODE REVIEW

File: pool.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func (d *Dispatcher) Wait() ([]Result, Errors) {
 }
 
 func (d *Dispatcher) cleanUp() {
+	close(d.jobQueue)
```
Comment (brandoneprice31) [inline pool.go]: 🙌

File: worker.go [hunk 2]
```diff
@@ -15,12 +13,12 @@ type (
 
 // NewWorker creates a new worker that can be registered to a workerPool
 // and receive jobs
-func NewWorker(workerPool chan chan Job, errCh chan error, resultCh chan Result) Worker {
+//func NewWorker(workerPool chan chan Job, errCh chan error, resultCh chan Result) Worker {
```
Comment (bigodines) [inline worker.go]: remove old signature

File: worker.go [hunk 3]
```diff
@@ -29,16 +27,20 @@ func NewWorker(workerPool chan chan Job, errCh chan error, resultCh chan Result)
...
 		for {
-			// register the current worker into the worker queue.
-			w.workerPool <- w.jobCh
-
 			select {
-			case job := <-w.jobCh:
-				result, err := job.Execute()
-				if err != nil {
-					w.errCh <- err
+			case job, more := <-w.jobCh:
```
Comment (brandoneprice31) [inline worker.go]: nice, I didn't know about the second return arg

File: pool_test.go [hunk 4]
```diff
@@ -86,6 +88,24 @@ func TestInvalid(t *testing.T) {
...
+		t.Fail()
+	}
+
+	d.Run()
+	d.Enqueue(EchoJob{}, EchoJob{}, EchoJob{})
+	d.Wait()
+
+	// sleep so goroutines have time to exit
+	time.Sleep(1000 * time.Millisecond)
+	if ngr != runtime.NumGoroutine() {
```
Comment (brandoneprice31) [inline pool_test.go]: nit -

```suggestion
assert.Equal(t, ngr, runtime.NumGoroutine())
```

**Comment by brandoneprice31** [comment]
Wait the commit suggestion didn't clear out the lines below it lol

File: worker.go (hunk 3 above)
Comment (bctsui) [reply worker.go]: > nice, I didn't know about the second return arg

yea, pretty handy... it's false if there are no more items and the channel is closed

**Pull request merged**